xbmc/pvr/dialogs/GUIDialogPVRChannelGuide.cpp

```diff
@@ -70,23 +70,33 @@ bool CGUIDialogPVRGuideOSD::OnMessage(CGUIMessage& message)
   return CGUIDialog::OnMessage(message);
 }
 
-void CGUIDialogPVRGuideOSD::OnInitWindow()
+void CGUIDialogPVRChannelGuide::Open(const CPVRChannelPtr &channel)
 {
-  /* Close dialog immediately if no TV or radio channel is playing */
-  if (!g_PVRManager.IsPlaying())
-  {
-    Close();
-    return;
-  }
+  m_channel = channel;
+  CGUIDialog::Open();
+}
 
+void CGUIDialogPVRChannelGuide::OnInitWindow()
+{
   // lock our display, as this window is rendered from the player thread
   g_graphicsContext.Lock();
   m_viewControl.SetCurrentView(DEFAULT_VIEW_LIST);
 
   // empty the list ready for population
   Clear();
 
-  g_PVRManager.GetCurrentEpg(*m_vecItems);
+  // no user-specific channel set use current playing channel
+  if (!m_channel)
+    m_channel = g_PVRManager.GetCurrentChannel();
```

Comment:
I know it was like this before, but it is no good idea to call into g_PVRManager with locked g_graphicsContext. May I suggest to move the two lines above to the very beginning of this method, before locking the global graphics mutex?